linux/LibSensors.c

```diff
@@ -261,6 +261,43 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
       goto out;
    }
 
+   /* Read sysfs to get HT/SMT cores and correctly distribute the temperature values */
+   if (coreTempCount < activeCPUs) {
+      double oldData[existingCPUs + 1];
+      memcpy(oldData, data, sizeof(oldData));
+
+      int dataCell = 1;
+      for (unsigned int i = 0; i < existingCPUs; i++) {
+         char path[100];
+         sprintf(path, "/sys/devices/system/cpu/cpu%d/topology/thread_siblings_list", i);
```

Comment:
We have `xSnprintf` function in htop codebase. Please use that instead.

Reply:
Also if you are iterating over several items in a common parent directory, it might be worth using things like `opendir` and `openat`.